src/evict/evict_page.c

```diff
@@ -235,6 +237,10 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF_STATE previous_state, u
         goto done;
     }
 
+    /* Check if the page has updates before reconciling */
+    if (page->modify != NULL)
```

Comment:
Instead of checking for updates, how about checking whether the page is modified and incrementing the dirty stat based on that flag? The reason behind this suggestion is that, even for the dirty page, we can have the updates.